PyInstaller/depend/imphook.py

```diff
@@ -374,8 +377,20 @@ def _load_hook_module(self):
         Class docstring for supported attributes.
         """
 
-        # If this hook script module has already been loaded, noop.
-        if self._hook_module is not None:
+        # If this hook script module has already been loaded,
+        # or we are _shallow, noop.
+        if self._hook_module is not None or self._shallow:
+            if self._shallow:
+                self._hook_module = True  # Not None
+                # Inform the user
+                logger.debug(
+                    'Skipping module hook %r from %r because a hook for %s has'
+                    ' already been loaded.',
+                    *os.path.split(self.hook_filename)[::-1], self.module_name
+                )
+                # Set the default attributes to empty instances of the type.
+                for attr_name, (attr_type, _) in _MAGIC_MODULE_HOOK_ATTRS:
+                    setattr(self, attr_name, attr_type())
```

Comment:
Causes an infinite recursion.
```suggestion
                    super(ModuleHook, self).__setattr__(attr_name, attr_type())
```

Reply:
Good catch